src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusService.scala

```diff
@@ -67,18 +70,34 @@ class ConsignmentStatusService(consignmentStatusRepository: ConsignmentStatusRep
   }
 
   def updateConsignmentStatus(updateConsignmentStatusInput: ConsignmentStatusInput): Future[Int] = {
-    validateStatusTypeAndValue(updateConsignmentStatusInput)
-    consignmentStatusRepository.updateConsignmentStatus(
-      updateConsignmentStatusInput.consignmentId,
-      updateConsignmentStatusInput.statusType,
-      updateConsignmentStatusInput.statusValue,
-      Timestamp.from(timeSource.now)
-    )
+    val statusType: String = updateConsignmentStatusInput.statusType
+
+    if (statusesToUpdateBasedOnFile.contains(statusType)) {
+      updateBasedOnFileStatus(updateConsignmentStatusInput.consignmentId, statusType)
+    } else {
+      validateStatusTypeAndValue(updateConsignmentStatusInput)
+      consignmentStatusRepository.updateConsignmentStatus(
+        updateConsignmentStatusInput.consignmentId,
+        updateConsignmentStatusInput.statusType,
+        updateConsignmentStatusInput.statusValue.get, // if this value is None, then it should through an error as this should not be possible after the validation
+        Timestamp.from(timeSource.now)
+      )
+    }
+  }
+
+  private def updateBasedOnFileStatus(consignmentId: UUID, statusType: String): Future[Int] = {
+    val noFileStatusesError = s"Error: There are no $statusType statuses for any of the files from consignment $consignmentId"
```

Comment:
Maybe re-word error message slightly to: `... statuses for any files from consignment ...`